JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -312,6 +312,32 @@ module Microsoft.ApplicationInsights {
         }
 
         /**
+         * Sets the autheticated user id and the account id in this session.
+         * User auth id and account id should be of type string. They should not contain commas, semi-colons, equal signs, spaces, or vertical-bars.
+         *   
+         * @param authenticatedUserId {string} - The authenticated user id. A unique and persistent string that represents each authenticated user in the service.
+         * @param accountId {string} - An optional string to represent the account associated with the authenticated user.
+         */
+        public setAuthenticatedUserContext(authenticatedUserId: string, accountId?: string) {
```

Comment:
In a previous thread, @Assaf-Neufeld mentioned that one of the scenarios we want to support is when the user switches between accounts, which brings me back to my earlier question regarding splitting the set method. Having used languages with first-class property concept for a long time, I'd also find working with two methods per property, like `getAuthenticatedUserId` and `clearAuthenticatedUserId`, confusing. Since we are already discussing the merits of encapsulating `User.authId` property with `User.getAuthId` method, what do you think about the single-method convention for encapsulating properties? 

```TypeScript
public authenticatedUserId(value?: string) {
  // if typeof value === undefined: return current value
  // else: store new value in cookie
  //    special case: if value === "", remove current value from cookie
}

// Get example
var currentAuthenticatedUserId = appInsights.authenticatedUserId();

// Set example
appInsights.authenticatedUserId("JohnDoe@mycompany.com");

// Clear example
appInsights.authenticatedUserId("");
```

This is common property pattern in C++ and also in some JavaScript frameworks, like [knockout](http://knockoutjs.com/documentation/observables.html). In this case we would still have only two methods, `authenticatedUserId()` and `userAccountId`.